### PR TITLE
Add exercise generator

### PR DESCRIPTION
**Pull request by mienaikage**

### CODE REVIEW

**Comment by kotp** [comment]
Does it make sense to separate this out, a PR for the exercise generator and individual ones for each exercise?  This could mirror "how it should be" when new exercises come in?

**Comment by mienaikage** [comment]
I can separate these out. The main reason I'd included them in here was so there were files to use with the generator if anyone wanted to test it out.

**Comment by kotp** [comment]
Yeah, I thought about that as well.  One that works as an example to work with it definitely makes sense.

**Comment by mienaikage** [comment]
hello-world yaml now included.

File: bin/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+The `exercise-gen.pl6` file must currently be run from within an exercise directory. It will retrieve data from an `example.yaml` file within that directory, and use the contained information to generate a test file using `templates/test.mustache`. If it finds a `canonical-data.json` file in `x-common` for the exercise in question it will be included.
```
Comment (kotp) [inline bin/README.md]: Does it matter which exercise directory?  If so, the general "from within an exercise directory" may be confusing.

If possible, it should be able to be generated while sitting in the project root (this is to say, where `.git` folder sits in the project).

Comment (mienaikage) [inline bin/README.md]: I've updated the docs to be a little clearer. The generator does not yet do multiple exercises, it simply uses the `example.yaml` file in the directory you're currently in, and figures out what canonical data you need for the exercise based on the directory name.

**Comment by mienaikage** [comment]
I've added new functionality to the generator. You can now specify args for the exercises you wish to generate, or give the arg `--all` to generate all exercises with an `example.yaml` file

**Comment by kotp** [comment]
Thank you @mienaikage for this...

**Pull request merged**